web/api/jobsets.py

```diff
@@ -82,40 +83,44 @@ def delete_jobset(
     return {"message": "Jobset deleted"}
 
 
+class EvaluationUpload(BaseModel):
+    git_revision: str
+    definition: dict
+
+
 @router.put("/{jobset_id}/upload-evaluation", response_model=schemas.EvaluationResponse)
 def upload_evaluation(
     jobset_id: int,
-    definition: dict,
+    body: EvaluationUpload,
     user_id: int = Depends(get_user),
     db: Session = Depends(get_db),
 ):
     """Upload a new evaluation for a jobset"""
-    from .. import models
-
-    eval = crud.create_evaluation(db, jobset_id, definition)
-    # parse the sbom and populate derivation list
-    components = definition["components"]
+    # Check if evaluation with this revision already exists
+    existing = crud.get_evaluation_by_revision(db, jobset_id, body.git_revision)
+    if existing:
+        raise HTTPException(
+            status_code=409,
+            detail=f"Evaluation with revision '{body.git_revision}' already exists for this jobset"
```

Comment:
this makes sense to me - we could have different reports for different parameters/architectures/etc but those could be separate jobsets 👍